src/opensynth/models/faraday/model.py

```diff
@@ -499,33 +511,47 @@ def train_gmm(self, dm: LCLDataModule):
         Args:
             dm (LCLDataModule): Training data
         """
+        logger.info("🚀 Training GMM")
+
         dl = dm.train_dataloader()
         next_batch = next(iter(dl))
 
         # Explicit check to make sure that data module used
         # to train GMM has features ordered the same way as
         # when training the VAE
-        obtained_feature_list = list(next_batch["features"].keys())
+        features = next_batch["features"]
+        obtained_feature_list = list(features.keys())
         expected_feature_list = self.vae_module.feature_list
-        assert obtained_feature_list == expected_feature_list
-
-        for batch_num, batch_data in tqdm(enumerate(dl)):
-            kwh = batch_data["kwh"]
-            features = batch_data["features"]
-
-            vae_input = self.vae_module.reshape_data(kwh, features)
-            vae_output = self.vae_module.encode(vae_input)
+        if obtained_feature_list != expected_feature_list:
+            logger.error(
```

Comment:
using logging instead of `assert`. I think assert is only meant to be used when debugging and not for catching/throwing errors inside production code.